### ISSUE TITLE
confusion over scales parameter for sample_energy/photon_flux

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#799]
The documentation for the `scales` parameter for `sample_energy_flux` (everything also holds for `sample_photon_flux`) claims that when `correlated=False` the `scales` parameter should be a 1D vector with `n` elements, and when `correlated=True` it is a 2D `n x n` matrix. This matches the underlying `sherpa.sim.sample.ParameterScaleVector/Matrix` classes that eventually get called (through a string of classes and methods). That is, these classes expect the errors (1D) or covariance matrix (2D). We do need to improve the documentation to indicate that one is an error and one a variance.

The reason for this ticket however is that `sample_energy_flux` fails if `correlated=False` and `scales` is a 1D NumPy array. That is:

```python
sherpa.astro.ui.sample_energy_flux(lo=0.5, hi=7, num=1, scales=np.asarray([0.04, 0.11, 2.4e-5]))
```

fails with the error

```
~/sherpa/sherpa-sample-flux/sherpa/astro/flux.py in sample_flux(fit, data, src, method, correlated, num, lo, hi, numcores, samples)
     90     if correlated == False and samples is not None:
     91         if numpy.ndarray == type(samples):
---> 92             samples = samples.diagonal(0)
     93
     94     samples = sampler.get_sample(fit, samples, num=num)

ValueError: diag requires an array of at least two dimensions
```

The work around is to use a Python list, since this works:

```python
sherpa.astro.ui.sample_energy_flux(lo=0.5, hi=7, num=1, scales=[0.04, 0.11, 2.4e-5])
```

The error comes because the code doesn't distinguish between 1D and 2D NumPy arrays, which it could. Or we could require a 1D array when `correlated=False` (as suggested by the docs).

**Comment by anetasie** [comment]
@DougBurke for documentation - 1D array is a 1 sigma width of the assumed Gaussian distribution for each model parameter.  By default this should be given by the covar() output. An option to supply the array of values would be needed when somebody wants to update these values.

**Comment by DougBurke** [comment]
I've separated out the fixes into #803 (from #754), along with a number of other inter-related changes.
